File: software/examples/rust/example_threshold.rs
use std::{error::Error, io, thread};
use tinkerforge::{current25_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Current25 Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let c = Current25Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get threshold receivers with a debounce time of 10 seconds (10000ms).
    c.set_debounce_period(10000);

    let current_reached_receiver = c.get_current_reached_callback_receiver();

    // Spawn thread to handle received callback messages.
    // This thread ends when the `c` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for current_reached in current_reached_receiver {
            println!("Current: {} A", current_reached as f32 / 1000.0);
        }
    });

    // Configure threshold for current "greater than 5 A".
    c.set_current_callback_threshold('>', 5 * 1000, 0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/current25_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Current25 Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	c, _ := current25_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get threshold receivers with a debounce time of 10 seconds (10000ms).
	c.SetDebouncePeriod(10000)

	c.RegisterCurrentReachedCallback(func(current int16) {
		fmt.Printf("Current: %f A\n", float64(current)/1000.0)
	})

	// Configure threshold for current "greater than 5 A".
	c.SetCurrentCallbackThreshold('>', 5*1000, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
